tests/pipeline/test_pipeline.py

```diff
@@ -167,32 +162,6 @@ def update_state_dict(key: str):
     assert state_dict["sink_start_async"]
 
 
-@pytest.mark.use_cudf
-def test_pipeline_narrowing_types(config: Config, filter_probs_df: DataFrameType):
-    """
-    Test to ensure that we aren't narrowing the types of messages in the pipeline.
-
-    In this case, `ConvMsg` emits `MultiResponseMessage` messages which are a subclass of `MultiMessage`,
-    which is the accepted type for `MultiMessagePassThruStage`. We want to ensure that the type is retained allowing us
-    to place a stage after `MultiMessagePassThruStage` requring `MultiResponseMessage` like `AddScoresStage`.
-    """
-    config.class_labels = ['frogs', 'lizards', 'toads', 'turtles']
-    expected_df = filter_probs_df.to_pandas()
-    expected_df = expected_df.rename(columns=dict(zip(expected_df.columns, config.class_labels)))
-
-    pipe = LinearPipeline(config)
-    pipe.set_source(InMemorySourceStage(config, [filter_probs_df]))
-    pipe.add_stage(DeserializeStage(config, ensure_sliceable_index=True))
-    pipe.add_stage(ConvMsg(config))
-    # pipe.add_stage(MultiMessagePassThruStage(config))
-    pipe.add_stage(AddScoresStage(config))
-    pipe.add_stage(SerializeStage(config, include=[f"^{c}$" for c in config.class_labels]))
-    compare_stage = pipe.add_stage(CompareDataFrameStage(config, compare_df=expected_df))
-    pipe.run()
-
-    assert_results(compare_stage.get_results())
-
-
```

Comment:
Lets update this test to use new dummy classes and ensure we are checking that stages dont narrow the type

Reply:
```python
    pipe = Pipeline(config)

    def my_source() -> DerivedControlMessage:
        yield DerivedControlMessage()

    pipe.add_stage(my_source(config))

    # Add a normal map stage that does ControlMessage -> ControlMessage

    @stage
    def my_sink(msg: DerivedControlMessage) -> DerivedControlMessage:
        return msg
```